app/controllers/avo/actions_controller.rb

```diff
@@ -9,9 +9,14 @@ class ActionsController < ApplicationController
       # set_record will fail if it's tried to be used from the Index page.
       request.params[:id].present?
     end
+    before_action :set_query, only: [:show, :handle], if: ->(request) do
+      # set query will fail in show if user will not select some row
+      # params[:fields] are set during selecting row or in show view
+      request.params[:fields].present?
+    end
```

Comment:
This approach makes sense for `show`, but we should ensure that the conditional check does not apply to `handle`.